compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/DecomposeWinogradPass.cpp

```diff
@@ -82,6 +82,94 @@ struct FoldWinogradOpUnitDims : public OpRewritePattern<TransformOp> {
   }
 };
 
+/// Pattern to decompose the tiled WinogradFilterTransformOp.
+/// The input should be just a single tile of the input filter of size `m x m`,
+/// where `m` is the kernel size. The output size should be `i x i`, where
+/// `i = m + r - 1` and `r` is the output tile size.
+///
+/// The tile of the filter transform decomposes into two matrix multiplications:
+/// `matmul(G, matmul(tile(f), transpose(G)))`
+/// The matrix `G` is a precomputed constant.
+///
+/// The result of the decomposition will look like the following:
+/// ```
+/// %tf = iree_linalg_ext.winograd.filter_transform
+///     output_tile_size(6) kernel_size(3) kernel_dimensions([0, 1])
+///     ins(%in_tile : tensor<3x3xf32>)
+///     outs(%out_tile : tensor<8x8xf32>) -> tensor<8x8xf32>
+/// ```
+/// Decomposes to
+/// ```
+/// %G = arith.constant dense<[...]>
+/// %GT = arith.constant dense<[...]>
+/// %init_0 = linalg.fill ins(%zero) outs(%out_tile) -> tensor<3x8xf32>
+/// %mm_0 = linalg.matmul ins(%in_tile, %GT : tensor<3x3xf32>, tensor<3x8xf32>)
+///     outs(%init_0 : tensor<3x8xf32>) -> tensor<3x8xf32>
+/// %init_1 = linalg.fill ins(%zero) outs(%out_tile) -> tensor<8x8xf32>
+/// %mm_1 = linalg.matmul ins(%G, %mm_0 : tensor<8x3xf32>, tensor<3x8xf32>)
+///     outs(%init_1 : tensor<8x8xf32>) -> tensor<8x8xf32>
+/// ````
+struct DecomposeWinogradFilterTransform
+    : public OpRewritePattern<WinogradFilterTransformOp> {
+  using OpRewritePattern<WinogradFilterTransformOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(WinogradFilterTransformOp transformOp,
+                                PatternRewriter &rewriter) const override {
+    Location loc = transformOp.getLoc();
+    auto funcOp = transformOp->getParentOfType<mlir::FunctionOpInterface>();
+    if (!funcOp) {
+      return rewriter.notifyMatchFailure(transformOp,
+                                         "Could not find parent function");
+    }
+    rewriter.setInsertionPointToStart(&funcOp.getFunctionBody().front());
```

Comment:
Are these needed? I think we can create all operations in-place, so we don't need to move insertion point?

Reply:
The funcOp check looks redundant to me